src/ray/raylet/node_manager.cc

```diff
@@ -1335,10 +1322,34 @@ void NodeManager::ProcessAnnounceWorkerPortMessage(
 
   auto message = flatbuffers::GetRoot<protocol::AnnounceWorkerPort>(message_data);
   int port = message->port();
+  RAY_LOG(ERROR) << "NodeManager::ProcessAnnounceWorkerPortMessage " << port
+                 << ", assigned " << worker->AssignedPort() << ", is worker" << is_worker;
   worker->Connect(port);
   if (is_worker) {
     worker_pool_.OnWorkerStarted(worker);
     HandleWorkerAvailable(worker);
+  } else {
+    auto job_data_ptr = worker->GetDriverJobTableData();
+    job_data_ptr->mutable_driver_address()->set_port(port);
+    RAY_CHECK_OK(gcs_client_->Jobs().AsyncAdd(job_data_ptr, [client](Status status) {
+      if (!status.ok()) {
+        RAY_LOG(ERROR) << "Failed to add job to GCS: " << status.ToString();
+        return;
+      }
```

Comment:
This is not the current behavior I think. The new code means we will never reply back to the driver if adding job fails?

Reply:
not sure what we need to do if gcs add failed. Maybe we should just check fail. But yeah I will warn-and-continue for now

Reply:
Let's just keep the old behavior (no regression) in this PR. We can think about what the better behavior is in follow-ups.